src/grid.h

```diff
@@ -519,24 +531,27 @@ struct Grid {
     void InsertCells(int dx, int dy, int nxs, int nys, Cell *nc = nullptr) {
         assert(((dx < 0) == (nxs == 0)) && ((dy < 0) == (nys == 0)));
         assert(nxs + nys == 1);
-        Cell **ocells = cells;
-        cells = new Cell *[(xs + nxs) * (ys + nys)];
+        vector<Cell *> ocells = std::move(cells);
+        cells = vector<Cell *>((xs + nxs) * (ys + nys));
         xs += nxs;
         ys += nys;
-        Cell **ncp = ocells;
+        auto oit = ocells.begin();
         SetOrient();
-        foreachcell(c) if (x == dx || y == dy) {
-            if (nc)
-                c = nc;
-            else {
-                Cell *colcell = ocells[(nxs ? max(0, min(dx - 1, xs - nxs - 1)) : x) +
-                                       (nxs ? y : max(0, min(dy - 1, ys - nys - 1))) * (xs - nxs)];
-                c = new Cell(cell, colcell);
-                c->text.relsize = colcell->text.relsize;
+        foreachcell(c) {
+            if (x == dx || y == dy) {
+                if (nc) {
+                    c = nc;
+                } else {
+                    Cell *colcell =
+                        ocells[(nxs ? max(0, min(dx - 1, xs - nxs - 1)) : x) +
+                               (nxs ? y : max(0, min(dy - 1, ys - nys - 1))) * (xs - nxs)];
+                    c = new Cell(cell, colcell);
+                    c->text.relsize = colcell->text.relsize;
+                }
+            } else {
+                c = *oit++;

```

Comment:
This is new, why was that not needed before?

Reply:
Ok, applied the old style